QLC+ 5: Add linear regression to BPM tapping

Wazzledi opened the pull request:
As described in forum thread https://www.qlcplus.org/forum/viewtopic.php?p=71374, I added the linear regression method used in QLC+ 4 to the BPM tapping functionality of QLC+ 5.

mcallegari commented on qmlui/qml/KeyPad.qml:
```diff
@@ -95,18 +101,60 @@ Rectangle
+                        while (tapHistory.length > 16) tapHistory.splice(0,1)
+
+                        //copy tap history to sort it
+                        for(var i = 0; i < tapHistory.length; i++)
```
space after `for`
But how about this?
`tapHistorySorted = tapHistory.slice();`

Wazzledi replied:
Adjusted!
Yes, this is far better.

mcallegari commented on qmlui/qml/KeyPad.qml:
```diff
@@ -95,18 +101,60 @@ Rectangle
+                        //init needed variables
+                        var n = 1, tapx = 0, tapy = 0, sum_x = 0, sum_y = 0, sum_xx = 0, sum_xy = 0
+                        
+                        for(var i = 0; i < tapHistory.length; i++)
```
space after `for`

Wazzledi replied:
Added the space.

mcallegari commented:
Hey, how about the Time edit tool too?
https://github.com/mcallegari/qlcplus/blob/master/qmlui/qml/TimeEditTool.qml

There's another time tapping

Wazzledi commented:
I totally forgot about this thing.
For that, I moved the algorithm in the TimeUtil.js and added it to the TimeEditTool.
While testing the TimeEditTool, I noticed that the Tap button is also shown when the chaser is set to beat.
Is that intended? Also, I dont see x/2 and 2x anymore to get half beat chasers.

mcallegari commented:
> For that, I moved the algorithm in the TimeUtil.js and added it to the TimeEditTool. 
Excellent. I would have done the same!

> While testing the TimeEditTool, I noticed that the Tap button is also shown when the chaser is set to beat. Is that intended? 
Nope. It's actually not needed

> Also, I dont see x/2 and 2x anymore to get half beat chasers.
I guess you mean like Speed Dial? TimeEditTool is not meant to be a replacement for Speed Dial

Wazzledi commented:
> I guess you mean like Speed Dial? TimeEditTool is not meant to be a replacement for Speed Dial
When editing the chaser step length with the TimeEditTool and the type is set to beat, I assumed the buttons x/2 and 2x, which are present in the code, should be present.
If they are not intended to be visible, then just ignore this comment.

mcallegari commented:
They are visible on a Beat-type RGB Matrix

![image](https://github.com/mcallegari/qlcplus/assets/2951337/a8cae0a5-39a0-445b-86da-687e6e472ff4)

mcallegari commented:
Many thanks for this :+1:
